multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakUserManager.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.multiuser.keycloak.server;
+
+import static java.lang.System.currentTimeMillis;
+import static java.util.Collections.emptyList;
+import static org.eclipse.che.commons.lang.NameGenerator.generate;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.util.Optional;
+import javax.inject.Named;
+import org.eclipse.che.account.api.AccountManager;
+import org.eclipse.che.api.core.ApiException;
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.model.user.User;
+import org.eclipse.che.api.core.notification.EventService;
+import org.eclipse.che.api.user.server.event.BeforeUserRemovedEvent;
+import org.eclipse.che.api.user.server.event.PostUserPersistedEvent;
+import org.eclipse.che.api.user.server.model.impl.UserImpl;
+import org.eclipse.che.api.user.server.spi.PreferenceDao;
+import org.eclipse.che.api.user.server.spi.ProfileDao;
+import org.eclipse.che.api.user.server.spi.UserDao;
+import org.eclipse.che.multiuser.api.account.personal.PersonalAccountUserManager;
+
+/**
+ * Extension of User Manager, providing utility operations related to Keycloak User management, and
+ * overriding create/remove operations to be compatible with {@link
+ * org.eclipse.che.multiuser.keycloak.server.dao.KeycloakProfileDao}
+ *
+ * @author Mykhailo Kuznietsov
+ */
+public class KeycloakUserManager extends PersonalAccountUserManager {
```

Comment:
Consider add `@Singleton` annotation or bind `KeycloakUserManager` in singleton scope